gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ObservationMapper.java

```diff
@@ -93,45 +98,75 @@ public List<Observation> mapResources(RCMRMT030101UKEhrExtract ehrExtract, Patie
+        var observation = buildBaseObservation(ehrComposition, observationStatement, patient, practiseCode, id, meta);
+
+        addContextToObservation(observation, encounters, ehrComposition);
+        addValue(observation, getValueQuantity(observationStatement.getValue(), observationStatement.getUncertaintyCode()),
+            getValueString(observationStatement.getValue()));
+        addEffective(observation,
+            getEffective(observationStatement.getEffectiveTime(), observationStatement.getAvailabilityTime()));
+
+        return observation;
+    }
+
+    private @NotNull Observation buildBaseObservation(RCMRMT030101UKEhrComposition ehrComposition,
+                                                RCMRMT030101UKObservationStatement observationStatement,
+                                                Patient patient,
+                                                String practiseCode,
+                                                String id,
+                                                Meta meta) {
+        var observation = new Observation()
             .setStatus(FINAL)
             .addIdentifier(buildIdentifier(id, practiseCode))
             .setCode(getCode(observationStatement.getCode()))
             .setIssuedElement(getIssued(ehrComposition))
             .addPerformer(getParticipantReference(observationStatement.getParticipant(), ehrComposition))
             .setInterpretation(getInterpretation(observationStatement.getInterpretationCode()))
             .setComment(getComment(
-                    observationStatement.getPertinentInformation(),
-                    observationStatement.getSubject(),
-                    observationStatement.getCode(),
-                    Optional.of(Optional.ofNullable(observationStatement.getCode())
+                observationStatement.getPertinentInformation(),
+                observationStatement.getSubject(),
+                observationStatement.getCode(),
+                Optional.of(Optional.ofNullable(observationStatement.getCode())
                             .map(CD::getQualifier)
                             .orElse(Collections.emptyList()))
             ))
             .setReferenceRange(getReferenceRange(observationStatement.getReferenceRange()))
-            .setSubject(new Reference(patient));
-        observation.setId(id);
-        observation.setMeta(generateMeta(META_PROFILE));
-
-        addContextToObservation(observation, encounters, ehrComposition);
-        addValue(observation, getValueQuantity(observationStatement.getValue(), observationStatement.getUncertaintyCode()),
-            getValueString(observationStatement.getValue()));
-        addEffective(observation,
-            getEffective(observationStatement.getEffectiveTime(), observationStatement.getAvailabilityTime()));
+            .setSubject(new Reference(patient))
+            .setMeta(meta)
+            .setId(id);
 
-        return observation;
+        return (Observation) observation;
```

Comment:
Just wondering if this casting is required? Looks like on line 137 a new Observation is being instantiated and that's what's being returned after setting additional members?

Reply:
I am trying to utilize the Fluent API in here and as the last calls (.setMeta(meta) and setId(id)) return Resource supertype I am casting it back to Observation as otherwise it alerts a bug.

Reply:
Ah gotcha, no problem!